src/serviceWorker/document.ts

```diff
@@ -22,9 +24,9 @@ export function matchDocument({ request, url }: RouteMatchCallbackOptions) {
     return false
   }
 
-  // If this isn't app.uniswap.org (or a local build), skip.
-  // IPFS gateways may not have domain separation, so they cannot use document caching.
-  if (url.hostname !== 'app.uniswap.org' && !isDevelopment()) {
+  // If this isn't a DNS domain skip. IPFS gateways may not have domain
+  // separation, so they cannot use App Shell-style routing.
+  if (!DNS_DOMAINS.includes(url.hostname) && !url.hostname.endsWith(DEV_DOMAINS_SUFFIX) && !isDevelopment()) {
```

Comment:
not sure if your want to use this https://github.com/cowprotocol/cowswap/blob/2b50ae0d6cc24511f6cad5aebe983e2085190374/src/custom/utils/environments.ts#L15


https://github.com/cowprotocol/cowswap/blob/2b50ae0d6cc24511f6cad5aebe983e2085190374/.env#L38

Reply:
Maybe not, just to make this more self contain and simpler...

Reply:
It's a copy of the original uniswap files.
Anyway, we have enough modifications in this file and we can use our helpers here.